### PR TITLE
Fix scripting error in model metadata sidebar

### PR DESCRIPTION
**Pull request by daltojohnso**
This seems to be breaking the model metadata sidebar on stats. See: https://metaboat.slack.com/archives/C03SHMZ266R/p1664815460770359?thread_ts=1663701716.003909&cid=C03SHMZ266R. I'm unable to repro locally for whatever reason.

The columns on a query and the columns on a card can differ slightly, specifically in the case of a column from a joined table. In the above scenario there's a column that lacks a `join-alias` in its `field_ref` when coming from a query, but not when it is coming from a card. Is this the result of relying on a query from `/api/dataset`?

Regardless, we shouldn't be filtering out columns when when we can't find metadata on the query results so that we can at least show data in the sidebar.

### CODE REVIEW

**Comment by kulyk** [comment]
> The columns on a query and the columns on a card can differ slightly, specifically in the case of a column from a joined table. In the above scenario there's a column that lacks a join-alias in its field_ref when coming from a query, but not when it is coming from a card. Is this the result of relying on a query from /api/dataset?

Most likely; the reason we used to re-run the query on entering/leaving the model editor is to prevent that from happening.

When a model is open in the chill-mode, we use `/api/dataset` and query `{ source-table: 'card__${MODEL_ID}' }`. In this case, the returned `result_metadata` would look slightly different as opposed to what we'd get from `/api/card/:id/query`:

* `join-alias` would be missing
* custom columns would just be field literals like `[ "field", "Custom", { "base-type": "..." } ]`

FWIW these two PRs should have a bit more context: #19952, #19721

This might be a sign that `/api/dataset` metadata squeezes into the model editor somehow 🤔

File: frontend/src/metabase/query_builder/components/DatasetEditor/DatasetEditor.jsx
Comment (kulyk) [inline frontend/src/metabase/query_builder/components/DatasetEditor/DatasetEditor.jsx:204]: AFAIR `orderedColumns` in this case are coming from viz settings and only contain a column name and a field ref. Not sure if we should really use them here?

**Review by gusaiani — APPROVED** [review]
👍

**Comment by mazameli** [comment]
@daltojohnso, are you waiting on a review from @kulyk?

**Comment by daltojohnso** [comment]
@mazameli -- yes, I think it is probably worthwhile for him to take a look at this code before I merge it.

**Review by kulyk — APPROVED** [review]
Looks good! Apologies for a delay

**Pull request merged**
